standard_sequence/classy_sequence.py

```diff
@@ -863,18 +1075,33 @@ def __init__(self):
     # if shot_globals.do_tweezer_position_check:
     #     t = do_tweezer_position_check(t)
 
-    if shot_globals.do_tweezer_check:
+    elif shot_globals.do_tweezer_check:
         TweezerSequence_obj = TweezerSequence(t)
         t = TweezerSequence_obj._do_tweezer_check_sequence(t)
 
     # if shot_globals.do_tweezer_check_fifo:
     #     t = do_tweezer_check_fifo(t)
 
-    if shot_globals.do_optical_pump_in_tweezer_check:
+    elif shot_globals.do_optical_pump_in_tweezer_check:
         TweezerSequence_obj = TweezerSequence(t)
         t = TweezerSequence_obj._do_optical_pump_in_tweezer_check(t)
 
     # if shot_globals.do_optical_pump_in_microtrap_check:
     #     t = do_optical_pump_in_microtrap_check(t)
 
+    """ Here doing all the finish up quirk for spectrum cards """
+    try:
+        current_obj = MOTSeq_obj
+    except:
+        try:
+            current_obj = OPSeq_obj
+        except:
+            try:
+                current_obj = TweezerSequence_obj
+            except:
+                raise NotImplementedError
+            t = current_obj.TweezerLaser_obj.stop_tweezers(t)
+
+    t = current_obj.Microwave_obj.reset_spectrum(t)
+
```

Comment:
The depth of nesting here is a little confusing. It might be better to say try: current_obj = X; except: pass multiple times rather than entering a new try inside an except statement. Then it would be more clear what current_obj is.

Reply:
This would not work greatly since if the method have not been implemented then there is no error raised. Let's stay with it now unless there is a better structure that can reduce the depth and allow the error raising at the same time.